controllers/ovncontroller_controller.go

```diff
@@ -263,11 +264,45 @@ func (r *OVNControllerReconciler) reconcileNormal(ctx context.Context, instance
 
 	instance.Status.Conditions.MarkTrue(condition.InputReadyCondition, condition.InputReadyMessage)
 
+	dbmap, err := v1beta1.GetDBEndpoints(ctx, helper, instance.Namespace, map[string]string{})
+	if err != nil {
+		// TODO(ihar) is it a valid return while waiting for dbcluster?
+		return ctrl.Result{}, err
+	}
+
+	// TODO(ihar) pull the "SB" string from a const package
+	sbCluster, err := v1beta1.GetDBClusterByType(ctx, helper, instance.Namespace, map[string]string{}, "SB")
+	if err != nil {
+		// TODO(ihar) is it a valid return while waiting for dbcluster?
+		return ctrl.Result{}, err
+	}
+	// TODO(ihar) check that sbCluster is not nil (or change the interface of the ByType function)
+
+	// TODO(ihar) move the code below to a separate helper function
+	// TODO(ihar) introduce a constructor for internal-SB string
+	externalConfig := make(map[string]string)
+	// TODO(ihar) should we fail on missing key? perhaps the interface of GetDbEndpoints should be more stringent
+	val, ok := dbmap["internal-SB"]
+	if ok {
+		// TODO(ihar) define keys here and elsewhere in a const file
+		externalConfig["ovn-remote"] = val
+	}
+
+	// TODO(ihar) do we need ovn- prefixes for all of the settings?
```

Comment:
those are just internal keys used in this operator, nothing public, right?
If so, I think we can get rid of the "ovn-" prefix there.

Reply:
These are actually - in a way - the public interface for external tools (in case of EDP this is Ansible) that would like to integrate with service operators. From this perspective, I consider them public.

The question about prefix is more about "cleanliness" and avoiding duplication (since the `ConfigMap` name itself contains the `ovn-` prefix). If a tool wants to have `ovn` as part of their path to configuration options, it can always mount the `ConfigMap` to a directory that has `ovn` in its full path.

Reply:
While we are thinking in terms of tight coupling between EDPM Ansible and service operators, the fact that this is an interface is of little relevance. What I'm thinking of is - a potential future where EDPM Ansible is no longer used, perhaps because there's a better / alternative way to deploy Compute nodes (as OCP nodes?) In this case, this interface will become shared / reused. This is e.g. why I don't set the keys in the `data` section of the spec with the names used by EDPM Ansible roles (with `edpm_` prefixes etc.) - I don't want this interface to assume a particular way of deploying Compute nodes.

Reply:
I am still not sure how ansible would be using it. I assume ansible itself wouldn't care about how it get's data i.e from config map or some other sources. May be some operator like dataplane/ansibleee operator would prepare the data in required format for ansible usage?

For prefix part i think it should be fine to keep the prefix as that's what is what expected by target system(ovs config external-ids).

Reply:
@karelyatin dataplane-operator will mount the ConfigMap data into edpm-ansible environment, see https://docs.google.com/document/d/1tr5KzgT0cB-nCF36kSmEbzxy9i38oLQyfCsGwgQUiY8/edit

That said, I now figured that I should transform this from direct ConfigMap to data: <file>: format, and then dataplane-operator will mount <file> into a file path. This PR will change direction somewhat to accommodate for that.